dashboard/config/routes.rb

```diff
@@ -466,10 +466,16 @@
             get 'page/:puzzle_page', to: 'script_levels#show', as: 'puzzle_page', format: false
             # /s/xxx/lessons/yyy/levels/zzz/sublevel/sss
             get 'sublevel/:sublevel_position', to: 'script_levels#show', as: 'sublevel', format: false
-            # /s/xxx/lessons/yyy/levels/zzz/summary
-            get 'summary', to: 'script_levels#show', defaults: {view: 'summary'}
           end
         end
+        resources :script_levels, only: [:show], path: "/levels", format: false do
+          # This route is defined in a separate resources, below the one above,
+          # because of how our assert_routing tests and Rails routing
+          # precedence work with multiple routes that point to the same action,
+          # with only a static path (no dynamic parts like 'path/:id').
+          # /s/xxx/lessons/yyy/levels/zzz/summary
+          get 'summary', on: :member, to: 'script_levels#show', as: 'summary', format: false, defaults: {view: 'summary'}
```

Comment:
this took forever to figure out, and seems super hacky, since it's only to support the tests :/

Reply:
Thanks for figuring it out! And thanks for the comment :)